src/lib/dom/base.js

```diff
@@ -299,3 +299,76 @@ function equal(bundles1, bundles2) {
   return bundles1.length === bundles2.length &&
     bundles1.every(({lang}, i) => lang === bundles2[i].lang);
 }
+
+// A regexp to sanitize HTML tags and entities.
+const reHtml = /[&<>]/g;
+const htmlEntities = {
+  '&': '&amp;',
+  '<': '&lt;',
+  '>': '&gt;',
+};
+
+// Unicode bidi isolation characters.
+const FSI = '\u2068';
+const PDI = '\u2069';
+
+/**
+ * Sanitize string-typed arguments.
+ *
+ * Escape HTML tags and entities and wrap values in the Unicode Isolation Marks
+ * (FSI and PDI) to ensure the proper directionality of the interpolated text.
+ *
+ * @param   {Object} args
+ * @returns {Object}
+ * @private
+ */
+function sanitizeArgs(args) {
+  for (const name in args) {
+    const arg = args[name];
+    if (typeof arg === 'string') {
+      const value = arg.replace(reHtml, match => htmlEntities[match]);
+      args[name] = `${FSI}${value}${PDI}`;
+    }
+  }
+  return args;
+}
+
+/**
+ * A bound version of `keysFromContext` using `entityFromContext`.
+ *
+ * @param {MessageContext} ctx
+ * @param {Array<Array>}   keys
+ * @param {{
+ *   errors: Array<Error>,
+ *   hasErrors: boolean,
+ *   translations: Array<{value: string, attrs: Object}>
+ * }} prev
+ * @returns {{
+ *   errors: Array<Error>,
+ *   hasErrors: boolean,
+ *   translations: Array<{value: string, attrs: Object}>
+ * }}
+ * @private
+ */
+function entitiesFromContext(ctx, keys, prev) {
+  return keysFromContext(entityFromContext, sanitizeArgs, ctx, keys, prev);
```

Comment:
Can you explain what you mean? Isn't what I did here a partially applied function?